File: topics/dynamic-memory/application/print-all-rows.c
void print_all_rows(const data_store *db_data)
{
    row *current;
    
    if (db_data == NULL) return;
    
    current = db_data->first_row;
    
    // While there is a current node
    while(current != NULL)
    {
        // Print the row to the Terminal
        print_row(*current);        // Follow the pointer, pass the value
        current = current->next;    // Move to next Row
    }
}

File: topics/dynamic-memory/application/PrintAllRows.pas
procedure PrintAllRows(const dbData: DataStore);
var
    current: ^Row;
begin
    current := dbData^.firstRow;
    
    // While there is a current node
    while current <> nil do
    begin
        // Print the row to the Terminal
        PrintRow(current^);        // Follow the pointer, pass the value
        current := current^.next;  // Move to next Row
    end;
end;
